### DISCUSSION TITLE
Error __typenames for type-safe error handling

### DISCUSSION THREAD

**Discussion by kevinpijning** [nuwave/lighthouse#2538] [Q&A] [answered]
I was having some issues regarding type-safe frontend graphql error handling and stubled on to the following article:
https://the-guild.dev/blog/graphql-error-handling-with-fp#graphql-default-error-handling

**Situation**
I wanted to implement the form validation (or any other ClientAware exceptions) of lighthouse in a typescript frontend project, but the error's returned from graphql do not have a type, its just json? 

**Question**
Is it possible to archieve the type declaration from the article above without creating custom resolvers for everything, or what are the best practices to handle errors from graphql in a type-safe front-end environment using lighthouse-php?

**Comment by spawnia** [comment, marked as answer]
Lighthouse does not provide facilities for returning errors as part of the payload. You can achieve it with custom field resolvers or field middleware.
